src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/ReactorAccessPort.java

```diff
@@ -1,192 +1,240 @@
+							inv.replaceExistingItem(slot, null);
+						}
+					}
+					for (int slot : getOutputSlots()) {
+						if (inv.getItemInSlot(slot) != null) {
+							b.getWorld().dropItemNaturally(b.getLocation(), inv.getItemInSlot(slot));
+							inv.replaceExistingItem(slot, null);
+						}
+					}
+				}
+				return true;
+			}
+		});
+	}
+	
+	private void constructMenu(BlockMenuPreset preset) {
+		for (int i : border) {
+			preset.addItem(i, new CustomItem(new ItemStack(Material.GRAY_STAINED_GLASS_PANE), " "),
+				(p, slot, item, action) -> false
+			);
+		}
+		
+		for (int i : border_1) {
+			preset.addItem(i, new CustomItem(new ItemStack(Material.LIME_STAINED_GLASS_PANE), " "),
+				(p, slot, item, action) -> false
+			);
+		}
+		
+		for (int i : border_2) {
+			preset.addItem(i, new CustomItem(new ItemStack(Material.CYAN_STAINED_GLASS_PANE), " "),
+				(p, slot, item, action) -> false
+			);
+		}
+		
+		for (int i : border_3) {
+			preset.addItem(i, new CustomItem(new ItemStack(Material.GREEN_STAINED_GLASS_PANE), " "),
+				(p, slot, item, action) -> false
+			);
+		}
+		
+		preset.addItem(1, new CustomItem(SlimefunItems.URANIUM, "&7Fuel Slot", "", "&rThis Slot accepts radioactive Fuel such as:", "&2Uranium &ror &aNeptunium"),
+			(p, slot, item, action) -> false
+		);
+		
+		preset.addItem(22, new CustomItem(SlimefunItems.PLUTONIUM, "&7Byproduct Slot", "", "&rThis Slot contains the Reactor's Byproduct", "&rsuch as &aNeptunium &ror &7Plutonium"),
+			(p, slot, item, action) -> false
+		);
+		
+		preset.addItem(7, new CustomItem(SlimefunItems.REACTOR_COOLANT_CELL, "&bCoolant Slot", "", "&rThis Slot accepts Coolant Cells", "&4Without any Coolant Cells, your Reactor", "&4will explode"),
+			(p, slot, item, action) -> false
+		);
+		
+		preset.addItem(7, new CustomItem(SlimefunItems.REACTOR_COOLANT_CELL, "&bCoolant Slot", "", "&rThis Slot accepts Coolant Cells", "&4Without any Coolant Cells, your Reactor", "&4will explode"),
+			(p, slot, item, action) -> false
+		);
+	}
+	
+	public String getInventoryTitle() {
+		return "&2Reactor Access Port";
+	}
+
+	public int[] getInputSlots() {
+		return new int[] {19, 28, 37, 25, 34, 43};
+	}
+	
+	public int[] getFuelSlots() {
+		return new int[] {19, 28, 37};
+	}
+	
+	public int[] getCoolantSlots() {
+		return new int[] {25, 34, 43};
+	}
+	
+	public static int[] getOutputSlots() {
+		return new int[] {40};
+	}
+	
+	public AReactor getReactor(Location l) {
+		Location reactorL = new Location(l.getWorld(), l.getX(), l.getY() - 3, l.getZ());
+		SlimefunItem item = BlockStorage.check(reactorL.getBlock());
+		if(item instanceof AReactor)
+			return (AReactor) item;
+	
+		return null;
+	}
+	
+	public BlockMenu getReactorMenu(Location l) {
+		Location reactorL = new Location(l.getWorld(), l.getX(), l.getY() - 3, l.getZ());
+		
+		SlimefunItem item = BlockStorage.check(reactorL);
+		
+		if(item != null && (item.getID().equals("NUCLEAR_REACTOR") || item.getID().equals("NETHERSTAR_REACTOR")))

```

Comment:
You don't need "item", you should use BlockStorage.checkID instead, parsing it as a SlimefunItem is a heavy call and not needed here, since you are only comparing the ID anyway.

Reply:
Updated to use BlockStorage.checkID